### PR TITLE
Fixed remove last dns provider

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: frontend/src/composables/useShootDns.js [hunk 1]
```diff
@@ -45,6 +45,21 @@ export const useShootDns = (manifest, options) => {
...
+      syncProvidersFromShootSpecDNS: false,
+    }
+    if (providers) {
+      providerConfig.providers = providers
+    }
+    setExtension({
+      type: 'shoot-dns-service',
+      providerConfig,
+    })
+  }
```
Comment (holgerkoser) [inline frontend/src/composables/useShootDns.js:61]:
```suggestion
  function setDnsServiceExtensionProviders (providers) {
    setExtension({
      type: 'shoot-dns-service',
      providerConfig: {
        apiVersion: 'service.dns.extensions.gardener.cloud/v1alpha1',
        kind: 'DNSConfig',
        syncProvidersFromShootSpecDNS: false,
        providers,
      },
    })
  }
```
It make no difference if the property does not exist or the value is `undefined`

File: frontend/src/composables/useShootDns.js [hunk 2]
```diff
@@ -177,15 +192,7 @@ export const useShootDns = (manifest, options) => {
...
-      setExtension({
-        type: 'shoot-dns-service',
-        providerConfig: {
-          apiVersion: 'service.dns.extensions.gardener.cloud/v1alpha1',
-          kind: 'DNSConfig',
-          syncProvidersFromShootSpecDNS: false,
-          providers: [provider],
-        },
-      })
+      setDnsServiceExtension([provider])
```
Comment (holgerkoser) [inline frontend/src/composables/useShootDns.js:195]:
```suggestion
      setDnsServiceExtensionProviders([provider])
```

File: frontend/src/composables/useShootDns.js [hunk 3]
```diff
@@ -222,6 +229,9 @@ export const useShootDns = (manifest, options) => {
       // Migrate from old DNS configuration to new DNS configuration
       set(dnsServiceExtension.value, 'providerConfig.syncProvidersFromShootSpecDNS', false)
       dnsProviders.value = filter(dnsProviders.value, ['primary', true])
+    } else if (!dnsServiceExtension.value) {
+      // set default empty dns service extension where sync is set to off
+      setDnsServiceExtension()
```
Comment (holgerkoser) [inline frontend/src/composables/useShootDns.js:234]:
```suggestion
      setDnsServiceExtensionProviders()
```

File: frontend/src/composables/useShootDns.js (hunk 2 above)
Comment (grolu) [reply frontend/src/composables/useShootDns.js:195]: why would you rename the function. I agree that it makes it clearer to include `providers` in the name as they are passed. But this function not only sets the providers but also the extension itself. I think that this information gets lost by removing it?

Comment (holgerkoser) [reply frontend/src/composables/useShootDns.js:195]: The only information you pass to this function is the list of providers. The rest is hardcoded and can not be changed. So it does not update the extension but only the provider. As as side effect it populates the hardcoded rest of the extension properties. Both solutions are not optimal. But I wanted to keep the propsal simple.

Comment (holgerkoser) [inline frontend/src/composables/useShootDns.js:195]:
```suggestion
      setDnsServiceExtension({ providers: [provider] })
```

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**